### PR TITLE
Feature/lu 7311 ingest baw trigger

### PR DESCRIPTION
**Pull request by mattprice-28**
New queue and lambda trigger added, lambda modified to get info from sqs event

### CODE REVIEW

File: aws-infrastructure/ingest.tf [hunk 1]
```diff
@@ -285,6 +285,21 @@ resource "aws_sqs_queue" "selection" {
         }
     )
 }
+
+# BAW ingest selection file trigger
+
+resource "aws_sqs_queue" "bawingesttrigger" {
+    name = "${var.environment_name}-bawingesttrigger"
```
Comment (dllmorgan) [inline aws-infrastructure/ingest.tf:292]: Use the standardised naming convention for new resources: ${local.name_prefix}

File: lambda-ingest-selection-file/handler.py [hunk 2]
```diff
@@ -6,14 +6,15 @@ def lambda_handler(event, context):
...
-    # json_filename = event['Records'][0]['s3']['object']['key']
-    # print(json_filename)
+    # For sqs trigger
+    
+    baw_notification = json.loads(event['Records'][0]['body'])
+    period           = baw_notification['period']
+    survey           = baw_notification['survey']
+    location         = baw_notification['location']
+    
+    name = location.split('.com/')
```
Comment (dllmorgan) [inline lambda-ingest-selection-file/handler.py]: The split will be on the / and not .com when using the S3 URI [s3://bucket-name/file]

**Review by dllmorgan — CHANGES_REQUESTED** [review]

File: aws-infrastructure/ingest.tf (hunk 1 above)
Comment (mattprice-28) [reply aws-infrastructure/ingest.tf:292]: Just looked through it, needs to be changed in quite a few places

File: lambda-ingest-selection-file/handler.py (hunk 2 above)
Comment (mattprice-28) [reply lambda-ingest-selection-file/handler.py]: I changed it to zone/ (assuming the bucket name convention stays the same), i was worried just doing / would create a split on the initial //

**Review by dllmorgan — APPROVED** [review]

**Review by ranjit-das — APPROVED** [review]
Tested and works as per acceptance criteria. (End-to-end test with BAW can't be done at the moment because integration with BAW story will be coming later.)

**Pull request merged**